main.go

```diff
@@ -240,6 +252,9 @@ func main() {
 			os.Exit(1)
 		}
 	}
+	if enableInstanceMetadataServiceLookup {
+		common.EnableInstanceMetadataServiceLookup()
```

Comment:
Is this a few feature?

Reply:
we should have done this earlier, most of the OCIservices do this. In order to construct the full URL of the OCI service, we need to lookup the instance metadata for few realms such as new realms or gov regions etc. If regions belong to file https://github.com/oracle/oci-go-sdk/blob/master/common/regions.json , the url constructed will be proper, else oc1 URL will be assumed. This flag will allow the region metdata lookup, similar to this - https://github.com/oracle/oci-cloud-controller-manager/blob/4102275236cb8644c24eaa07292e5a0eef9b2ac7/pkg/cloudprovider/providers/oci/ccm.go#L155